jerry-core/parser/regexp/re-compiler.c

```diff
@@ -614,25 +612,26 @@ re_compile_bytecode (const re_compiled_code_t **out_bytecode_p, /**< [out] point
 
     if (cache_idx == RE_CACHE_SIZE)
     {
-      if (re_cache_idx == 0u)
+      if (JERRY_CONTEXT (re_cache_idx) == RE_CACHE_SIZE)
       {
-        re_cache_idx = RE_CACHE_SIZE;
+        JERRY_CONTEXT (re_cache_idx) = 0;
       }
 
-      const re_compiled_code_t *cached_bytecode_p = re_cache[--re_cache_idx];
-      JERRY_DDLOG ("RegExp cache is full! Remove the element on idx: %d\n", re_cache_idx);
+      JERRY_DDLOG ("RegExp cache is full! Remove the element on idx: %d\n", JERRY_CONTEXT (re_cache_idx));
+
+      cache_idx = JERRY_CONTEXT (re_cache_idx)++;
 
-      if (cached_bytecode_p != NULL)
+      /* The garbage collector might run during the byte code
+       * allocations above and it may free this entry. */
+      if (JERRY_CONTEXT (re_cache)[cache_idx] != NULL)
       {
-        ecma_bytecode_deref ((ecma_compiled_code_t *) cached_bytecode_p);
+        ecma_bytecode_deref ((ecma_compiled_code_t *) JERRY_CONTEXT (re_cache)[cache_idx]);
```

Comment:
style of `JERRY_CONTEXT (...)[...]` is mixed. There is a space after `)` in some place. I prefer this style (without space). Please unify that.